### ISSUE TITLE
Batched requests don't handle errors

### ISSUE THREAD

**Issue by rkachowski** [mana-ethereum/ethereumex#95]
If a single request fails in a batched http request, it will throw the following error 

```
** (FunctionClauseError) no function clause matching in anonymous fn/1 in Ethereumex.HttpClient.format_batch/1
    (ethereumex 0.8.0) lib/ethereumex/client/base_client.ex:472: anonymous fn(%{"error" => %{"code" => -32000, "message" => "execution reverted"}, "id" => 86, "jsonrpc" => "2.0"}) in Ethereumex.HttpClient.format_batch/1
    (elixir 1.12.3) lib/enum.ex:1582: Enum."-map/2-lists^map/1-0-"/2
    (elixir 1.12.3) lib/enum.ex:1582: Enum."-map/2-lists^map/1-0-"/2
    (ethereumex 0.8.0) lib/ethereumex/http_client.ex:34: Ethereumex.HttpClient.decode_body/2
```

This is because the batch is [implicitly assumed to be completely successful](https://github.com/mana-ethereum/ethereumex/blob/master/lib/ethereumex/client/base_client.ex#L472) and the results will be returned in a tuple `{:ok, results = []}`. 

Unfortunately handling failing requests is going to require an api change, as currently the only way to map the response back to the request from application code is via the index of the response within the result list - i.e. it's not possible to return anything other than a single list and keep the order intact.

Currently I handle this by including the error map in the response list, but this is pretty dirty and requires custom parsing.

**Comment by ayrat555** [comment]
@rkachowski can you please check if [this pr](https://github.com/mana-ethereum/ethereumex/pull/98) works for you?

if it does, I'll add the change to a new version

**Comment by ayrat555** [comment]
merged and released in 0.9.0

**Comment by rkachowski** [comment]
that looks awesome, thanks for the change :)
